### PR TITLE
DQIB-3042-mp-b2b-ui-tests-refactoring

### PR DESCRIPTION
**Pull request by vadym-uskov**

### CODE REVIEW

File: tests/ui/e2e/e2e_mp_b2b.robot [hunk 1]
```diff
@@ -645,7 +639,7 @@ Return_Management
     Zed: go to my order page:    ${lastPlacedOrder}
     Zed: trigger matching state of xxx merchant's shipment:    1    send to distribution
     Zed: trigger matching state of xxx merchant's shipment:    1    confirm at center
-    Zed: trigger matching state of xxx merchant's shipment:    1    Ship
```
Comment (AntonShopin) [inline tests/ui/e2e/e2e_mp_b2b.robot:648]: please revert all changes related to upper/lower case in order states. 
We had a bug that on local env and on stagings we have a diff in cases. It's fixed and will be released soon. 
All item states will be in the upper case

File: tests/ui/e2e/e2e_mp_b2b.robot [hunk 2]
```diff
@@ -807,36 +801,33 @@ Discounts
     Zed: grand total for the order equals:    ${lastPlacedOrder}    €18.11
     [Teardown]    Run keywords    Zed: login on Zed with provided credentials:    ${zed_admin_email}
     ...    AND    Zed: go to second navigation item level:    Merchandising    Discount
-    ...    AND    Zed: Deactivate Following Discounts From Overview Page:    Voucher Code 5% ${random}    Cart Rule 10% ${random}    Promotional Product 100% ${random}        
+    ...    AND    Zed: Deactivate Following Discounts From Overview Page:    Voucher Code 5% ${random}    Cart Rule 10% ${random}    Promotional Product 100% ${random}
```
Comment (AntonShopin) [inline tests/ui/e2e/e2e_mp_b2b.robot:804]: let's make sure that return discounts from the demodata back to it's initial state -> We need to activate: `Free chair    Tu & Wed $5 off 5 or more    10% off $100+    Free marker    20% off storage    	Free office chair    Free standard delivery    	10% off Safescan    5% off white    Tu & Wed €5 off 5 or more    10% off minimum order`

File: resources/common/common_zed.robot [hunk 3]
```diff
@@ -100,11 +100,9 @@ Zed: Uncheck Checkbox by Label:
     Uncheck Checkbox    xpath=//input[@type='checkbox']/../../label[contains(text(),'${checkbox_label}')]//input
 
 Zed: submit the form
-    wait until element is visible    ${zed_save_button}
-    # Wait Until Keyword Succeeds    3 x    3 s    Click    
-    Click Element by xpath with JavaScript    //input[contains(@class,'safe-submit')]
+    Wait until element is visible    ${zed_save_button}
+    Click    xpath=//input[contains(@class,'safe-submit')]    delay=1s
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply resources/common/common_zed.robot]: fixed

File: resources/steps/shopping_carts_steps.robot [hunk 4]
```diff
@@ -135,6 +135,7 @@ Yves: get link for external cart sharing
     Wait Until Element Is Visible    xpath=//input[@id='PREVIEW']
     ${externalURL}=    Get Element Attribute    xpath=//input[@id='PREVIEW']    value
     ${externalURL}=    Replace String Using Regexp    ${externalURL}    ${EMPTY}.*.com/    ${EMPTY}
+    ${externalURL}=    Replace String Using Regexp    ${externalURL}    ${EMPTY}.*.com/    ${EMPTY}
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply resources/steps/shopping_carts_steps.robot]: fixed

File: tests/ui/e2e/e2e_mp_b2b.robot (hunk 1 above)
Comment (vadym-uskov) [reply tests/ui/e2e/e2e_mp_b2b.robot:648]: fixed

File: resources/steps/products_steps.robot [hunk 5]
```diff
@@ -139,7 +139,7 @@ Zed: view abstract product page contains:
         END
         IF    '${key}'=='variants count' and '${value}' != '${EMPTY}'
             Clear Text    xpath=//div[@id='product-variant-table_filter']//input[@type='search']
-            Sleep    1s
+            Sleep    2s
```
[the remark this answers is not in the document]
Comment (vadym-uskov) [reply resources/steps/products_steps.robot]: fixed using 'Wait Until Element Is Visible'

**Pull request merged**